pandas/core/groupby/groupby.py

```diff
@@ -1275,8 +1275,14 @@ def std(self, ddof: int = 1):
         Series or DataFrame
             Standard deviation of values within each group.
         """
-        # TODO: implement at Cython level?
-        return np.sqrt(self.var(ddof=ddof))
+        result = self.var(ddof=ddof)
```

Comment:
I think rather than doing this you should just be able to call `_python_agg_general`, which should generically handle applying a function

Reply:
That won't get the cython speedup in the case when ddof=1 (and perhaps others, if the rest of var is cythonized), is that right?

Reply:
Hmm didn't realize we had a Cython implementation of var. I think can also use _cython_agg_general

Reply:
Can you elaborate? I'm guessing you're thinking of passing how="var" to _cython_agg_general. But I don't see what that gets us, other than duplicating the code in var.

Reply:
This comment should really apply to line 1279 below; we already have generic machinery in place to perform that looping depending on if we are working with a Series / DataFrame, so the branching and looping here shouldn’t be necessary

Reply:
Any chance you can comment back on this? I think this introduces code duplication as is

Reply:
@WillAyd Apologies, and I'll follow up on this PR if there is indeed some code duplication, but I don't think there is. What we're doing here is applying a composition of two functions, g composed with h, where h is cythonized. As far as I can tell, there is no way to pass this composition in and keep it cythonized. Also, there is no place to apply a post-processing function to a cythonized aggregation. If this is wrong, then the duplication should definitely be removed.

Reply:
Yea we also have a function called `_get_cythonized_result` which accepts a callable as an argument to `post_processing`

Reply:
Indeed I wasn't aware of this. Took a look into it, and what I'm finding is there are (at least) two different signatures for cython functions. The first is those used by _get_cythonized_result:

    result, labels, [vals], [mask], [ngroups]

and the second is that used in _aggregate (ignoring group_nth):

    result, counts, values, comp_ids, min_count

Here, comp_ids is the same as labels. Since var is used in _aggregate currently, the signature can't be changed to match what is expected in _get_cythonized_result without much hacking. I tried it and got it to work, but it seems to me that the two signature types should be sorted out first, otherwise the logic required for _get_cythonized_result is undesirable. If that sounds right, I'd like to raise an issue for this (I searched but couldn't find one).

Reply:
Yea I doubt there is an issue for it already as its really a matter of internal refactor. Things are a bit messy in the groupby code...so sure any clean up opportunities you see we would certainly welcome PRs for. Thanks for looking